File: 378-kth-smallest-element-in-a-sorted-matrix/378-kth-smallest-element-in-a-sorted-matrix.cpp
class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        vector<int> temp;
        for(auto i:matrix){
            for (auto j:i){
                temp.push_back(j);
            }
        }
        
        sort(temp.begin(),temp.end());
        return temp[k-1];
    }
};

File: 378-kth-smallest-element-in-a-sorted-matrix/378-kth-smallest-element-in-a-sorted-matrix.py
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:
        l,r = -10**9,10**9
        # ans = None
        while l<=r:
            m = (l+r)//2
            temp = 0
            for row in matrix:
                temp += bisect_right(row,m)
            if temp>=k:
                ans = m
                r = m-1
            else:
                l = m+1
        return l